src/main/java/baritone/process/CraftingProcess.java

```diff
@@ -0,0 +1,473 @@
+    }
+
+    private boolean hasCraftingTable() {
+        for (ItemStack itemstack : ctx.player().inventory.mainInventory) {
+            if (itemstack.getItem() == Item.getItemFromBlock(Blocks.CRAFTING_TABLE)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Goal createGoal(BlockPos pos) {
+        return new GoalGetToBlock(pos);
+    }
+
+    private boolean rightClick() { //shamelessly copied from go to block process
+        BlockPos bp = null;
+        if (goal instanceof GoalComposite) {
+            Goal[] goals = ((GoalComposite)goal).goals();
+            for (Goal goal : goals) {
+                if (goal.isInGoal(ctx.playerFeet()) && goal instanceof GoalGetToBlock) {
+                    bp = ((GoalGetToBlock)goal).getGoalPos();
+                    break;
+                }
+            }
+        } else {
+            bp = ((GoalGetToBlock)goal).getGoalPos();
+        }
+        if (bp != null) {
+            Optional<Rotation> reachable = RotationUtils.reachable(ctx.player(), bp, ctx.playerController().getBlockReachDistance());
+            if (reachable.isPresent()) {
+                baritone.getLookBehavior().updateTarget(reachable.get(), true);
+                if (bp.equals(ctx.getSelectedBlock().orElse(null))) {
+                    baritone.getInputOverrideHandler().setInputForceState(Input.CLICK_RIGHT, true);
+                    System.out.println(ctx.player().openContainer);
+                    if (!(ctx.player().openContainer instanceof ContainerPlayer)) {
+                        baritone.getInputOverrideHandler().clearAllKeys();
+                        return true;
+                    }
+                }
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private void placeCraftingtableNearby() {
+        selectCraftingTable();
+
+        ISchematic schematic = new FillSchematic(5, 4, 5, Blocks.CRAFTING_TABLE.getDefaultState());
+
+        List<IBlockState> desirableOnHotbar = new ArrayList<>();
+        Optional<Placement> toPlace = searchForPlaceables(schematic, desirableOnHotbar);
+
+        if (toPlace.isPresent()) {
+            Rotation rot = toPlace.get().rot;
+            baritone.getLookBehavior().updateTarget(rot, true);
+            ctx.player().inventory.currentItem = toPlace.get().hotbarSelection;
+            baritone.getInputOverrideHandler().setInputForceState(Input.SNEAK, true);
+            if ((ctx.isLookingAt(toPlace.get().placeAgainst) && ctx.objectMouseOver().sideHit.equals(toPlace.get().side)) || ctx.playerRotations().isReallyCloseTo(rot)) {
+                baritone.getInputOverrideHandler().setInputForceState(Input.CLICK_RIGHT, true);
+                placeCraftingTable = false;
+                getACraftingTable();
+            }
+        } else {
+            goal = new GoalRunAway(5, ctx.playerFeet());
+        }
+    }
+
+    private void selectCraftingTable() {
```

Comment:
done but this method name is not representative of what it is doing. its probably a historical reason why its called that but throwaway sounds like it will throw this items out of its inventory. something like hasAndOptionalSelect() would be more fitting.

Reply:
It's mande to select throwaway items, yes.

Reply:
It can take time though (inventory movement delay or waiting until stationary) so you should make sure you only use the item once it's actually there.